File: PlaybackLab/TVOSSampleApp/TVOSSampleApp/Players/AbstractPlayerViewController.m
//
//  AbstractPlayerViewController.m
//  TVOSSampleApp
//
//  Copyright © 2016 Ooyala. All rights reserved.
//

#import "AbstractPlayerViewController.h"
#import <OoyalaSDK/OOOoyalaPlayer.h>

@interface AbstractPlayerViewController ()

@end

@implementation AbstractPlayerViewController

- (void)notificationHandler:(NSNotification *)notification {
  if ([notification.name isEqualToString:OOOoyalaPlayerTimeChangedNotification]) {
    return;
  }
  
  NSLog(@"Notification Received: %@. state: %@. playhead: %f",
        notification.name,
        [OOOoyalaPlayerStateConverter playerStateToString:self.player.state],
        self.player.playheadTime);
}

@end

File: PlaybackLab/TVOSSwiftSampleApp/TVOSSwiftSampleApp/Players/AbstractPlayerViewController.swift
//
//  AbstractPlayerViewController.swift
//  TVOSSwiftSampleApp
//
//  Copyright © 2018 Ooyala. All rights reserved.
//

import Foundation

class AbstractPlayerViewController: UIViewController {
  
  var ooyalaPlayerViewController : OOOoyalaTVPlayerViewController!
  var option: PlayerSelectionOption!
  var player: OOOoyalaPlayer!
  
  @objc func notificationHandler(_ notification: Notification) {
    let name = notification.name
    if name == NSNotification.Name.OOOoyalaPlayerTimeChanged {
      return
    }
    
    NSLog("Notification Received: %@. state: %@. playhead: %f", name.rawValue,
          OOOoyalaPlayerStateConverter.playerState(toString: player.state),
          player.playheadTime())
  }
  
}
